### PR TITLE
Fix China AZs

### PR DESCRIPTION
**Pull request by paurosello**
China only has 2 AZs so we should open this a bit. In other regions, although we do not recommend it, 2AZ also works

### CODE REVIEW

**Review by calvix — APPROVED** [review]
lgtm if tested

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -112,8 +112,8 @@ func (f *flag) Validate() error {
 		// Validate Master AZs.
 		switch f.Provider {
 		case key.ProviderAWS:
-			if len(f.MasterAZ) != 0 && len(f.MasterAZ) != 1 && len(f.MasterAZ) != 3 {
-				return microerror.Maskf(invalidFlagError, "--%s must be set to either one or three availability zone names", flagMasterAZ)
+			if len(f.MasterAZ) < 1 && len(f.MasterAZ) > 3 {
```
Comment (axbarsan) [inline cmd/template/cluster/flag.go:115]: Not sure if the length can be both less than 1 and more than 3 at the same time 😬 

```suggestion
			if len(f.MasterAZ) < 1 || len(f.MasterAZ) > 3 {
```

**Review by axbarsan — CHANGES_REQUESTED** [review]

**Comment by axbarsan** [comment]
Please don't forget about updating `CHANGELOG.md` 👍🏻

**Comment by anvddriesch** [comment]
LGTM!
Just make sure this works correctly with 3 AZ names showing up in the CR like this:
```
availabilityZones:
- AZ1
- AZ2
- AZ2
```

**Comment by paurosello** [comment]
> LGTM!
> Just make sure this works correctly with 3 AZ names showing up in the CR like this:
> 
> ```
> availabilityZones:
> - AZ1
> - AZ2
> - AZ2
> ```

Oh true! we should not allow 2 AZ, just enable repeating AZ as the master needs to be either 1 or 3 always

**Review by axbarsan — APPROVED** [review]
LGTM after fixing CI 🌟

**Pull request merged**